packages/api/src/api/ProviderApi.ts

```diff
@@ -30,18 +32,30 @@ export default class ProviderApi extends HttpClientBase {
         return this.axios.get(`/v1/prosopo/providers`)
     }
 
-    public getCaptchaChallenge(
+    public async getCaptchaChallenge(
         userAccount: string,
-        randomProvider: ProsopoRandomProvider
+        randomProvider: ProsopoRandomProvider,
+        dappAccount: string,
+        retryAttempt = 0
     ): Promise<GetCaptchaResponse> {
         const { provider } = randomProvider
         const { blockNumber } = randomProvider
 
-        return this.axios.get(
-            `/v1/prosopo/provider/captcha/${provider.datasetId}/${userAccount}/${
-                this.network.dappContract.address
-            }/${blockNumber.toString().replace(/,/g, '')}`
-        )
+        if (retryAttempt > RETRY_LIMIT) {
+            throw new Error('Failed to get captcha challenge after maximum retry attempts')
+        }
+
+        try {
+            return this.axios.get(
+                `/v1/prosopo/provider/captcha/${provider.datasetId}/${userAccount}/${
+                    this.network.dappContract.address
+                }/${blockNumber.toString().replace(/,/g, '')}`
+            )
+        } catch (e) {
+            console.log(e)
+            const newRandomProvider = await this.getNewRandomProvider(userAccount)
+            return this.getCaptchaChallenge(userAccount, newRandomProvider, dappAccount, retryAttempt + 1)
```

Comment:
I wonder whether we should have some kind of sleep function here? E.g. sleep(1s). If the retry limit is 10 and the block time is 6s, I can see 10 retries all being done in the same block, all getting the same provider who isn't responding. If we add some kind of delay, e.g. 1 second, we would hit a block rollover and get another provider. Could do with discussing how long this delay should be / how many retries should be done.

I think we should do a fair number of retries but attempt to hit at least 3 blocks. That would give us 3 different providers, in theory. Although maybe 5 would be better.

Reply:
Yes agree. Block interval can be gotten from the API consts in polkadot

Reply:
`api.consts.babe?.expectedBlockTime`